presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -286,6 +288,36 @@ public static double stLength(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return geometry.getEsriGeometry().calculateLength2D();
     }
 
+    @SqlNullable
+    @Description("Returns a float between 0 and 1 representing the location of the closest point on the LineString to the given Point, as a fraction of total 2d line length.")
+    @ScalarFunction("line_locate_point")
+    @SqlType(DOUBLE)
+    public static Double lineLocatePoint(@SqlType(GEOMETRY_TYPE_NAME) Slice lineSlice, @SqlType(GEOMETRY_TYPE_NAME) Slice pointSlice)
+    {
+        if (lineSlice == null || pointSlice == null) {
+            return null;
+        }
+
+        Geometry line = JtsGeometrySerde.deserialize(lineSlice);
+        Geometry point = JtsGeometrySerde.deserialize(pointSlice);
+
+        if (line.isEmpty() || point.isEmpty()) {
+            return null;
+        }
+
+        GeometryType lineType = getForJtsGeometryType(line.getGeometryType());
+        if (lineType != LINE_STRING && lineType != MULTI_LINE_STRING) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("First argument to line_locate_point must be a LineString or a MultiLineString. Got: %s", line.getGeometryType()));
```

Comment:
Better add the entire value represented as JKT instead of just a type. It might be one of the values in your tables, and knowing what the value is will give you more insights.

Reply:
This is a good suggestion, but it needs to be applied broadly and WKT would need to be truncated to avoid super large messages. Perhaps, a separate PR?